packages/buidler-core/src/builtin-tasks/node.ts

```diff
@@ -63,6 +72,66 @@ Private Key: ${privateKey}
 }
 
 export default function () {
+  internalTask(TASK_NODE_WATCH_COMPILER_OUTPUT)
+    .addParam("url", "The URL of the node", undefined, types.string)
+    .setAction(async ({ url }: { url: string }, { config }) => {
+      const { default: fetch } = await import("node-fetch");
+      const chokidar = await import("chokidar");
+
+      const compilerVersion = config.solc.version;
+      const solcInputPath = path.join(config.paths.cache, SOLC_INPUT_FILENAME);
+      const solcOutputPath = path.join(
+        config.paths.cache,
+        SOLC_OUTPUT_FILENAME
+      );
+
+      const addCompilationResult = async () => {
+        if (
+          !(await fsExtra.pathExists(
+            path.join(config.paths.cache, SOLC_INPUT_FILENAME)
+          ))
+        ) {
+          return false;
+        }
+
+        if (
+          !(await fsExtra.pathExists(
+            path.join(config.paths.cache, SOLC_OUTPUT_FILENAME)
+          ))
+        ) {
+          return false;
+        }
+
+        const compilerInput = await fsExtra.readJSON(solcInputPath, {
+          encoding: "utf8",
+        });
+        const compilerOutput = await fsExtra.readJSON(solcOutputPath, {
+          encoding: "utf8",
+        });
+
+        await fetch(url, {
+          method: "POST",
+          body: JSON.stringify({
+            jsonrpc: "2.0",
+            method: "buidler_addCompilationResult",
+            params: [compilerVersion, compilerInput, compilerOutput],
+            id: 1,
+          }),
+        });
+      };
+
+      chokidar
+        .watch(solcOutputPath, {
+          ignoreInitial: true,
+          awaitWriteFinish: {
```

Comment:
This is necessary because sometimes `chokidar` emits the `change` event *before* the file is finished being written. This is even more likely in big projects, where the `solc-output.json` can be huge.